powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/database/model/converter/SignatureMetadataConverter.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * PowerAuth Server and related software components
+ * Copyright (C) 2023 Wultra s.r.o.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package io.getlime.security.powerauth.app.server.database.model.converter;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.getlime.security.powerauth.app.server.database.model.SignatureMetadata;
+import jakarta.persistence.AttributeConverter;
+import jakarta.persistence.Converter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * A JPA attribute converter for converting SignatureMetadata objects to and from JSON representations.
+ * This class enables storing SignatureMetadata in the database as a JSON column.
+ *
+ * @author Jan Dusil
+ */
+@Converter
+@Component
+@Slf4j
+public class SignatureMetadataConverter implements AttributeConverter<SignatureMetadata<?, ?>, String> {
```

Comment:
Why do we use the `<?, ?>` type specification?

Reply:
My general idea was that this converter could be used for different types of SignatureMetadata implementations not just the SignatureMetadata<String, String>. Since the props of implementations of SignatureMetadata could be different types then I thought the <?, ?> shall be here.

Reply:
In current code the usage of generics is really strange. See the following message from IntelliJ IDEA:
![image](https://github.com/wultra/powerauth-server/assets/26658588/023518f7-14ca-4ec5-b559-d18d6afdeaba)

My point is that we specify types for a generified class `SignatureMetadata`, however these types are never used, so they just make the code more complicated for no good reason. If you remove the `<T1, T2>` and `<?, ?>` types everywhere from the code for the `SignatureMetadata`, the tests are still green, so why do we need this generification in the first place?

Reply:
I guess that it's just a leftover from the previous version, where there used to be such code
```java
T1 getMetadataParam1();
```